triplea-game/triplea
# Remove support for MD5Crypt authentication in P2P games

## Overview

Removes support for MD5Crypt authentication in peer-to-peer network games (i.e. when connecting to a password-protected game server).  All stable releases in the wild for about a year now use HMAC-SHA512 authentication, so one more usage of the deprecated `Md5Crypt#hashPassword()` can be removed.

## Functional Changes

* Removed support for MD5Crypt authentication in peer-to-peer network games.

## Manual Testing Performed

Verified a client can connect to a password-protected server in the following scenarios:

* client: this branch, server: this branch
* client: this branch, server: 12226

:warning: Note that a 12226 client cannot connect to a server from this branch due to an oversight in the `ClientLogin#addAuthenticationResponseProperties()` method (that code should have actually tested to see if the challenge could be processed by the `Md5CryptAuthenticator` instead of simply assuming it could).  Since we've decided to break network compatibility, this shouldn't be a show-stopper, but consider this a heads-up.
